framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -93,6 +96,21 @@ private AnnotatedTypeMirror(TypeMirror underlyingType, AnnotatedTypeFactory atyp
     this.atypeFactory = atypeFactory;
   }
 
+  @Override
+  public AnnotatedTypeMirror clone() {
+    /// I cannot call a constructor, because AnnotatedTypeMirror is abstract.
+    // return new AnnotatedTypeMirror(
+    //     underlyingType, atypeFactory, underlyingTypeHashCode, annotations.clone());
+
+    try {
+      AnnotatedTypeMirror result = (AnnotatedTypeMirror) super.clone();
```

Comment:
AnnotatedTypeMirrors have to be carefully created so that any recursive types are correctly initialized.  So, I don't think this clone method will work.   Can you just call deepCopy instead?

Reply:
Good idea.  I didn't know deepCopy existed.  (I usually define clone and factory methods near the constructor.)